### PR TITLE
Offer list for organization - API endpoints

### PR DESCRIPTION
**Pull request by juliaanholcer**
__Description:__

Adding API methods that return a list of offers for an organization. New detail route for organizations viewset and filtering for offers (e.g. by organization.id).

__New imports / dependencies:__

* django_filter

__Unit Tests:__

* `apps/volontulo/tests/views/api/organizations/test_offers_list.py`
* some more detaild test will be written later

### CODE REVIEW

File: backend/apps/volontulo/tests/views/api/organizations/test_offers_list.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            '/api/organizations/{}/offers/'
+            .format(self.offer.organization.id))
+        self.assertEqual(res.status_code, 405)
+
+    def test_200(self):
+        """ Test for gettin offers list """
+        res = self.client.get(
+            '/api/organizations/{}/offers/'
+            .format(self.offer.organization.id))
+        self.assertEqual(res.status_code, 200)
```
Comment (magul) [inline backend/apps/volontulo/tests/views/api/organizations/test_offers_list.py]: can we test also content of the response here? see: https://github.com/CodeForPoznan/volontulo/blob/master/backend/apps/volontulo/tests/views/api/offers/test_list.py#L23

File: backend/apps/volontulo/views/api.py [hunk 2]
```diff
@@ -141,6 +142,17 @@ class OfferViewSet(viewsets.ModelViewSet):
 
     serializer_class = serializers.OfferSerializer
     permission_classes = (permissions.OfferPermission,)
+    filter_backends = (DjangoFilterBackend,)
+    filter_fields = (
+        'finished_at',
+        'location',
+        'organization',
```
Comment (magul) [inline backend/apps/volontulo/views/api.py]: what does it mean, to filter on `organization` here? we already filter on `organization__id` and `organization__name`.

File: backend/apps/volontulo/views/api.py [hunk 3]
```diff
@@ -175,3 +187,20 @@ def contact(request, pk):
             serializer.validated_data,
         )
         return Response(data=True, status=status.HTTP_201_CREATED)
+
+    @staticmethod
+    @detail_route(methods=['GET'], permission_classes=(AllowAny,))
+    # pylint: disable=invalid-name
+    def offers(request, pk):
+        """ Endpoint to get offers for organization """
+        get_object_or_404(Organization, id=pk)
```
Comment (magul) [inline backend/apps/volontulo/views/api.py]: What do You think about:
```
organization = get_object_or_404(Organization, id=pk)
if logged_as_admin(request):
    offers = organization.offer_set.get_for_administrator()
else:
    offers = organization.offer_set.get_weightened()
```

File: backend/apps/volontulo/views/api.py [hunk 4]
```diff
@@ -13,6 +13,7 @@
 from django.shortcuts import get_object_or_404
 from django.utils.encoding import force_bytes, force_text
 from django.utils.http import urlsafe_base64_encode, urlsafe_base64_decode
+from django_filters.rest_framework import DjangoFilterBackend
```
[the remark this answers is not in the document]
Comment (radoslawroszkowiak) [reply backend/apps/volontulo/views/api.py]: I think there should be introduced i.e. isort and linting to check and fix imports for us to avoid this kind of discussions in the future.

BTW, this order is correct, isn't it (django_filters > django)? ;)

File: backend/apps/volontulo/views/api.py (hunk 3 above)
Comment (radoslawroszkowiak) [reply backend/apps/volontulo/views/api.py]: I don't know the details of `get_for_administrator` and `get_weightened`, but I think rather than verifying permission levels in every single query (easy to forget somewhere, harder to refactor, etc.) maybe you could utilize query managers and apply permission filtering there. https://docs.djangoproject.com/en/2.0/topics/db/managers/

File: frontend/package-lock.json [hunk 5]
```diff
@@ -5005,7 +5005,7 @@
     "minimatch": {
       "version": "3.0.4",
       "resolved": "https://registry.npmjs.org/minimatch/-/minimatch-3.0.4.tgz",
-      "integrity": "sha1-UWbihkV/AzBgZL5Ul+jbsMPTIIM=",
+      "integrity": "sha512-yJHVQEhyqPLUTgt9B83PXu6W3rx4MvvHvSUvToogpwoGDOUQ+yDrR0HRot+yOCdCO7u4hX3pWft6kWBBcqh0UA==",
```
Comment (radoslawroszkowiak) [inline frontend/package-lock.json]: this change seems to be irrelevant to the API changes. I don't think this should be committed in this PR.

File: backend/apps/volontulo/views/api.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (juliaanholcer) [reply backend/apps/volontulo/views/api.py]: IMHO it is in alphabetic order.

[the remark this answers is not in the document]
Comment (magul) [reply backend/apps/volontulo/views/api.py]: yes, my bad here!

File: backend/apps/volontulo/views/api.py (hunk 2 above)
Comment (juliaanholcer) [reply backend/apps/volontulo/views/api.py]: There is also possible to use query 'organization=id' which is the same with 'organization__id=id'. I don't know whether we need both. Right now I'm not sure whether filtering works as I wanted, so I need to check it in details.

File: backend/apps/volontulo/views/api.py (hunk 3 above)
Comment (juliaanholcer) [reply backend/apps/volontulo/views/api.py]: @magul This works better because the previous version hasn't returned closed offers.

@radoslawroszkowiak These are methods for OffersManager class if I understand it correctly.

What is more important to me right now: for an administrator in organization details view all offers should be returned? for organization owner - also all offers and a possibility to edit them, but for other or unauthenticated users - only published? Who should see rejected offers?

File: backend/apps/volontulo/views/api.py (hunk 2 above)
Comment (juliaanholcer) [reply backend/apps/volontulo/views/api.py]: With organization, it is possible to filter offers with 'organization=id' query, but I don't know whether we need it. 'organization__id=id' gives the same result. For an organization-details view, I would prefer to utilize this second endpoint in OrganizationViewSet. Filtering here, for offers, can be used with all active and all archived offers views and later if we introduce some search functionality. We should decide what fields can be used for filtering purpose and add here.
